File: javascript/test/trajbell.js
#!/usr/bin/env node
var liba = require("../liba");
function assert(expr) {
    if (!eval(expr)) {
        throw Error(expr);
    }
}
function do_leak_check() {
    for (const arg of arguments) {
        arg.delete();
    }
    if (liba.do_leak_check) {
        liba.do_leak_check();
    }
}
var ctx = new liba.trajbell();
ctx.gen(3, 2, 3, 0, 10, 0, 0);
assert("ctx.t != undefined");
assert("ctx.tv != undefined");
assert("ctx.ta != undefined");
assert("ctx.td != undefined");
assert("ctx.taj != undefined");
assert("ctx.tdj != undefined");
assert("ctx.p0 != undefined");
assert("ctx.p1 != undefined");
assert("ctx.v0 != undefined");
assert("ctx.v1 != undefined");
assert("ctx.vm != undefined");
assert("ctx.jm != undefined");
assert("ctx.am != undefined");
assert("ctx.dm != undefined");
for (var i = 0; i < ctx.t; i += ctx.t / 10) {
    console.log(i, ctx.pos(i), ctx.vel(i), ctx.acc(i), ctx.jer(i));
}
do_leak_check(ctx);

File: python/test/trajbell.py
#!/usr/bin/env python
from __future__ import print_function
import os, sys

path = os.path.dirname(__file__)
path = os.path.dirname(path)
sys.path.insert(0, path)
if len(sys.argv) > 1:
    sys.stdout = open(sys.argv[1], "w")

import liba  # type: ignore

traj = liba.trajbell()
traj.gen(3, 2, 3, 0, 10)

N = 1000
T = traj.t / N
x = liba.new_real(N)
for i in range(N):
    x[i] = i * T

p = traj.pos(x)
v = traj.vel(x)
a = traj.acc(x)
j = traj.jer(x)

for i in range(N):
    print("%g,%g,%g,%g,%g" % (x[i], p[i], v[i], a[i], j[i]))

traj.t
traj.tv
traj.ta
traj.td
traj.taj
traj.tdj
traj.p0
traj.p1
traj.v0
traj.v1
traj.vm
traj.jm
traj.am
traj.dm
